typhos/utils.py

```diff
@@ -178,6 +193,17 @@ def __init__(self, *args, **kwargs):
         self._animation.setScaledSize(self._icon_size)
         self.setMovie(self._animation)
         self._animation.start()
+        if enable_timeout:
+            QtCore.QTimer.singleShot(self.LOADING_TIMEOUT_MS,
+                                     self._handle_timeout)
+
+    def _handle_timeout(self):
+        self._animation.stop()
+        self.setMovie(None)
+        self.setText("Loading Timeout")
+        self.setStyleSheet("color: red;")
+        self.setToolTip("Could not complete operation after "
+                        f"{self.LOADING_TIMEOUT_MS/1000} seconds.")
```

Comment:
Are there any problems here if the loading widget has already been removed/replaced? Or does QT clean up nicely for timers referencing methods in deleted widgets?

Reply:
On local tests it handled it properly. I started with an offline IOC and before the 10s I enabled the IOC, no issues were noticed.

Reply:
Seems like it's OK with a bound method:

```python
import qtpy
import qtpy.QtWidgets
import qtpy.QtCore


class Test(qtpy.QtWidgets.QWidget):
    def __init__(self):
        super().__init__()
        qtpy.QtCore.QTimer.singleShot(100, self.test)

        def test2():
            print('inner function')

        qtpy.QtCore.QTimer.singleShot(100, test2)

    def test(self):
        print('test called / bound method')


app = qtpy.QtWidgets.QApplication([])

test = Test()
test.deleteLater()
del test
test = None

app.exec_()
```

Results in: 
```
inner function
```

Reply:
```
test.deleteLater()
del test
test = None
```
It seems like you do not trust the Python garbage collector (probably warranted)